### PR TITLE
Accessing gradient of a coupled array variable's time derivative

### LINKED ISSUE #26385: Accessing gradient of a coupled array variable's time derivative
## Reason

Currently, the `Coupleable` object does not provide a function to access the gradient of a coupled array variable's time derivative.
This quantities are needed to implement the transient capability of HFEM-PN solver option in Griffin.

## Design

Add interface function in `Coupleable` to expose the gradient of a coupled array variable's time derivative.

## Impact

The gradient of a coupled array variable's time derivative can be accessible via `Coupleable`.

### PR DESCRIPTION
**Pull request by yjung-anl**
closes #26385

### CODE REVIEW

File: framework/include/interfaces/Coupleable.h [hunk 1]
```diff
@@ -818,6 +818,16 @@ class Coupleable
   virtual const ArrayVariableGradient & coupledArrayGradientOlder(const std::string & var_name,
                                                                   unsigned int comp = 0) const;
 
+  /**
+   * Retun gradient of a coupled array variable's time derivative
+   * @param var_name Name of coupled vector variable
```
Comment (lindsayad) [inline framework/include/interfaces/Coupleable.h:823]:
```suggestion
   * @param var_name Name of coupled array variable
```

File: framework/include/interfaces/Coupleable.h [hunk 2]
```diff
@@ -818,6 +818,16 @@ class Coupleable
...
+   * @param comp Component number for vector of coupled vector variables
```
Comment (lindsayad) [inline framework/include/interfaces/Coupleable.h:824]:
```suggestion
   * @param comp Component number for vector of coupled array variables
```

File: framework/include/interfaces/Coupleable.h [hunk 3]
```diff
@@ -818,6 +818,16 @@ class Coupleable
...
+   * @return Reference to a ArrayVariableGradient containing the time derivative of the gradient of
```
Comment (lindsayad) [inline framework/include/interfaces/Coupleable.h:825]: Is it the time derivative of the gradient, or the gradient of the time derivative? Above it's the latter but here it's the former

File: framework/src/interfaces/Coupleable.C [hunk 4]
```diff
@@ -1636,6 +1636,19 @@ Coupleable::coupledArrayGradientOlder(const std::string & var_name, unsigned int
   return var->gradSlnOlderNeighbor();
 }
 
+const ArrayVariableGradient &
+Coupleable::coupledArrayGradientDot(const std::string & var_name, unsigned int comp) const
+{
+  const auto * var = getArrayVar(var_name, comp);
```
Comment (lindsayad) [inline framework/src/interfaces/Coupleable.C:1642]:
```suggestion
  const auto * const var = getArrayVar(var_name, comp);
```

**Review by lindsayad — CHANGES_REQUESTED** [review]
Ideally we would add a test for this

**Comment by yjung-anl** [comment]
Comments have been addressed.

**Review by lindsayad — APPROVED** [review]

**Comment by lindsayad** [comment]
Thanks @yjung-anl !

**Pull request merged**